Marlin/src/feature/fwretract.cpp

```diff
@@ -99,8 +99,10 @@ void FWRetract::retract(const bool retracting
 
   // Simply never allow two retracts or recovers in a row
   if (retracted[active_extruder] == retracting) return;
-
   #if EXTRUDERS > 1
+    // Allow G10S1 after G10 'Priority for common sense'
+    if (swapping && (retracted_swap[active_extruder]== retracting))  return;
+    // G11 priority to recover the long retract if activated ' common sense '
```

Comment:
Very important , this will fix all issues here , about  zlifting never recover , because of the batching of simplify3d that movez Z between retraction ,   and Slic3r that place a G10 after a G10 S1 ;
For the lovers of Slic3r , the devs have made specially for marlin flavor , the modification to have firmware retractation compatible , but only if we use my update here